micro-rdk/src/common/grpc_client.rs

```diff
@@ -116,7 +118,33 @@ where
                     None => return Poll::Ready(None),
                 },
             };
-            self.buffer = chunk.into_data().unwrap();
+            self.buffer = match chunk.into_data() {
+                Ok(data) => data,
+                Err(e) => {
+                    if let Some(trailers) = e.trailers_ref() {
+                        if trailers.contains_key("grpc-message")
+                            && trailers.contains_key("grpc-status")
+                        {
+                            return Poll::Ready(Some(Err(GrpcClientError::GrpcError {
+                                code: trailers
+                                    .get("grpc-status")
+                                    .unwrap()
```

Comment:
Can we map these unwraps (and the ones immediately below) with one of the GrpcClientErrors?

Reply:
Good point but I don't want to clutter grpc error enum. Added defaults instead